### PR TITLE
Update connector

### PR DESCRIPTION
**Pull request by TomNong**
* Update `_mlp_transform` method.
* Polish connector.

### CODE REVIEW

**Comment by ZhitingHu** [comment]
Why did we open a new PR and lose all previous comments?

File: texar/torch/modules/connectors/connectors.py [hunk 1]
```diff
@@ -63,20 +63,10 @@ def _assert_same_size(outputs: TensorStruct,
         if isinstance(size, torch.Size):
             if output[0].size() != size:
                 raise ValueError("The output size does not match"
-                                 "the required output_size")
+                                 "the the required output_size")
```
Comment (huzecong) [inline texar/torch/modules/connectors/connectors.py]: Why add a `the`?

File: texar/torch/modules/connectors/connectors.py [hunk 2]
```diff
@@ -91,7 +81,7 @@ def _sum_output_size(output_size: OutputSize) -> int:
     if isinstance(flat_output_size[0], torch.Size):
         size_list = [0] * len(flat_output_size)
         for (i, shape) in enumerate(flat_output_size):
-            size_list[i] = np.prod([dim for dim in shape])
+            size_list[i] = np.prod(shape)
```
Comment (huzecong) [inline texar/torch/modules/connectors/connectors.py]: I thought we've talked about this. We can write this in a much more Pythonic way:
```python
size_list = [np.prod(shape) for shape in flat_output_size]
```

File: texar/torch/modules/connectors/connectors.py [hunk 3]
```diff
@@ -128,19 +113,15 @@ def _mlp_transform(inputs: TensorStruct,
...
     # Get output dimension
     flat_output_size = nest.flatten(output_size)
 
     if isinstance(flat_output_size[0], torch.Size):
         size_list = [0] * len(flat_output_size)
         for (i, shape) in enumerate(flat_output_size):
-            size_list[i] = np.prod([dim for dim in shape])
+            size_list[i] = np.prod(shape)
     else:
         size_list = flat_output_size
```
Comment (huzecong) [inline texar/torch/modules/connectors/connectors.py]: These few lines are exactly the same as a part of `_sum_output_size`. Why not take these out as a separate function (e.g. `_get_sizes`), and replace `_sum_output_size(xs)` with `sum(_get_sizes(xs))`?

**Comment by gpengzhi** [comment]
Any updates on this PR?

**Comment by AvinashBukkittu** [comment]
This PR is being tracked in #200

**Pull request closed without merging**